amr-wind/physics/multiphase/MultiPhase.cpp

```diff
@@ -520,4 +520,83 @@ void MultiPhase::levelset2vof()
     (*m_vof).fillpatch(0.0);
 }
 
+// Do levelset2vof with iblank neumann and into supplied scratch field
+void MultiPhase::levelset2vof(IntField& iblank_cell, ScratchField& vof_scr)
+{
+    const int nlevels = m_sim.repo().num_active_levels();
+    (*m_levelset).fillpatch(m_sim.time().current_time());
+    const auto& geom = m_sim.mesh().Geom();
+
+    for (int lev = 0; lev < nlevels; ++lev) {
+        auto& levelset = (*m_levelset)(lev);
+        auto& vof = vof_scr(lev);
+        const auto& dx = geom[lev].CellSizeArray();
+
+        for (amrex::MFIter mfi(levelset); mfi.isValid(); ++mfi) {
+            const auto& vbx = mfi.validbox();
+            const amrex::Array4<amrex::Real>& phi = levelset.array(mfi);
+            const amrex::Array4<amrex::Real>& volfrac = vof.array(mfi);
+            const amrex::Array4<const int>& iblank =
+                iblank_cell(lev).const_array(mfi);
+            const amrex::Real eps = 2. * std::cbrt(dx[0] * dx[1] * dx[2]);
+            amrex::ParallelFor(
+                vbx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                    // Neumann of levelset across iblank boundaries
+                    int ibdy = 0;
+                    int jbdy = 0;
+                    int kbdy = 0;
+                    if (iblank(i, j, k) != iblank(i - 1, j, k)) {
+                        ibdy = -1;
+                    }
+                    if (iblank(i, j, k) != iblank(i, j - 1, k)) {
+                        jbdy = -1;
+                    }
+                    if (iblank(i, j, k) != iblank(i, j, k - 1)) {
+                        kbdy = -1;
+                    }
+                    // no cell should be isolated such that -1 and 1 are
+                    // needed
+                    if (iblank(i, j, k) != iblank(i + 1, j, k)) {
+                        ibdy = +1;
+                    }
+                    if (iblank(i, j, k) != iblank(i, j + 1, k)) {
+                        jbdy = +1;
+                    }
+                    if (iblank(i, j, k) != iblank(i, j, k + 1)) {
+                        kbdy = +1;
+                    }
+                    amrex::Real mx, my, mz;
+                    multiphase::youngs_fd_normal_neumann(
+                        i, j, k, ibdy, jbdy, kbdy, phi, mx, my, mz);
+                    mx = std::abs(mx / 32.);
+                    my = std::abs(my / 32.);
+                    mz = std::abs(mz / 32.);
+                    amrex::Real normL1 = (mx + my + mz);
+                    mx = mx / normL1;
+                    my = my / normL1;
+                    mz = mz / normL1;
+                    // Make sure that alpha is negative far away from the
+                    // interface
+                    amrex::Real alpha;
+                    if (phi(i, j, k) < -eps) {
+                        alpha = -1.0;
+                    } else {
+                        alpha = phi(i, j, k) / normL1;
```

Comment:
why two lines? `alpha = phi(i, j, k) / normL1 + 0.5;` Should be fine? And then you use a ternary expression and can `const real alpha  =... `